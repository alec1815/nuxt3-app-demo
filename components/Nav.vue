<template>
  <div class="nav">
    <div class="nav-item">
      <NuxtLink to="/">商品</NuxtLink>
    </div>
    <span>|</span>
    <div>
      <NuxtLink to="/cart">购物车 (0)</NuxtLink>
    </div>
  </div>
</template>