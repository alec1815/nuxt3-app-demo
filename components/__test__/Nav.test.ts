import { mount } from '@vue/test-utils'
import Nav from '../Nav.vue'

describe('Nav', () => {
  it('should render correctly', () => {
    const wrapper = mount(Nav)
    expect(wrapper.element).toMatchSnapshot()
  })
})








// import { render, screen, fireEvent } from '@testing-library/vue'
// import Button from './Button.vue'

// describe('Button', () => {
//   it('has a label', () => {
//     render(Button, {
//       slots: { default: 'Hello' },
//     })

//     const button = screen.getByText('Hello')

//     expect(button).toBeInTheDocument()
//   })

//   it('can be clicked', () => {
//     const onClick = jest.fn()

//     render(Button, {
//       slots: { default: 'Hello' },
//       props: {
//         onClick,
//       },
//     })

//     expect(onClick).not.toHaveBeenCalled()

//     fireEvent.click(screen.getByText('Hello'))

//     expect(onClick).toHaveBeenCalledTimes(1)
//   })
// })
