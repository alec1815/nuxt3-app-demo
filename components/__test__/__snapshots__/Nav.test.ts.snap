// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render correctly 1`] = `
<div
  class="nav"
>
  <div
    class="nav-item"
  >
    <a>
      商品
    </a>
  </div>
  <span>
    |
  </span>
  <div>
    <a>
      购物车 (0)
    </a>
  </div>
</div>
`;
